runtime/compiler/x/codegen/J9TreeEvaluator.cpp

```diff
@@ -8072,113 +8071,90 @@ J9::X86::TreeEvaluator::VMarrayStoreCHKEvaluator(
 
       TR::Instruction* instr;
 
-#ifdef OMR_GC_COMPRESSED_POINTERS
-      // FIXME: Add check for hint when doing the arraystore check as below when class pointer compression
-      // is enabled.
+      if (TR::Compiler->om.compressObjectReferences())
+         {
 
-      TR::MemoryReference *destTypeMR = generateX86MemoryReference(destReg, TR::Compiler->om.offsetOfObjectVftField(), cg);
+         // FIXME: Add check for hint when doing the arraystore check as below when class pointer compression
+         // is enabled.
 
-      generateRegMemInstruction(L4RegMem, node, destComponentClassReg, destTypeMR, cg); // class pointer is 32 bits
-      TR::TreeEvaluator::generateVFTMaskInstruction(node, destComponentClassReg, cg);
+         TR::MemoryReference *destTypeMR = generateX86MemoryReference(destReg, TR::Compiler->om.offsetOfObjectVftField(), cg);
 
-      // -------------------------------------------------------------------------
-      //
-      // If the component type is java.lang.Object then the store always succeeds.
-      //
-      // -------------------------------------------------------------------------
+         generateRegMemInstruction(L4RegMem, node, destComponentClassReg, destTypeMR, cg); // class pointer is 32 bits
+         TR::TreeEvaluator::generateVFTMaskInstruction(node, destComponentClassReg, cg);
 
-      TR_OpaqueClassBlock *objectClass = fej9->getSystemClassFromClassName("java/lang/Object", 16);
+         // -------------------------------------------------------------------------
+         //
+         // If the component type is java.lang.Object then the store always succeeds.
+         //
+         // -------------------------------------------------------------------------
+
+         TR_OpaqueClassBlock *objectClass = fej9->getSystemClassFromClassName("java/lang/Object", 16);
 
-      if (TR::Compiler->target.is64Bit())
-         {
-#ifdef OMR_GC_COMPRESSED_POINTERS
```

Comment:
This seems fine since it's guarded above by a compressedrefs runtime check so the 64-bit non-compressedrefs path here is dead and the 32-bit path is the same as the compressedrefs path. Just pointing this out to other reviewers as it looked incorrect on my first review through this.